src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -474,5 +491,53 @@ void WiFiManager::PostConnectivityStatusChange(ConnectivityChange changeType)
     PlatformMgr().PostEventOrDie(&networkEvent);
 }
 
+System::Clock::Milliseconds32 WiFiManager::CalculateNextRecoveryTime()
+{
+    if (mConnectionRecoveryTimeMs > kConnectionRecoveryMaxIntervalMs)
+    {
+        // Find the new random jitter value in range [-jitter, +jitter].
+        int32_t jitter            = chip::Crypto::GetRandU32() % (2 * jitter + 1) - jitter;
+        mConnectionRecoveryTimeMs = kConnectionRecoveryMaxIntervalMs + jitter;
+        return System::Clock::Milliseconds32(mConnectionRecoveryTimeMs);
+    }
+    else
+    {
+        uint32_t currentRecoveryTimeout = mConnectionRecoveryTimeMs;
+        mConnectionRecoveryTimeMs       = mConnectionRecoveryTimeMs * 2;
+        return System::Clock::Milliseconds32(currentRecoveryTimeout);
+    }
+}
+
+void WiFiManager::Recover(System::Layer *, void *)
+{
+    // If kConnectionRecoveryMaxOverallInterval has a non-zero value prevent endless re-scan.
+    if (0 != kConnectionRecoveryMaxRetries && (++Instance().mConnectionRecoveryCounter >= kConnectionRecoveryMaxRetries))
+    {
+        Instance().AbortConnectionRecovery();
+        return;
+    }
+
+    // Prevent executing Scan if it has been aborted by the failsafe timeout occurrence.
+    if (!Instance().mRecoveryTimerAborted)
```

Comment:
This must be true, because the timer is cancelled when `mRecoveryTimerAborted` is set, right?

Reply:
You mean that the Recover will not be called all at all when the `AbortConnectionRecovery` is called hence the check is redundant?

Reply:
Yes, exactly.